packages/block-library/src/term-template/edit.js

```diff
@@ -226,14 +169,9 @@ export default function TermTemplateEdit( {
 		if ( ! terms ) {
 			return [];
 		}
-		if ( termsToShow === 'top-level' ) {
-			return terms.filter( ( term ) => ! term.parent );
-		}
-		if ( termsToShow === 'subterms' ) {
-			return terms.filter( ( term ) => term.parent );
-		}
-		return terms;
-	}, [ terms, termsToShow ] );
+		// Limit to the number of terms defined by perPage.
+		return perPage === 0 ? terms : terms.slice( 0, perPage );
```

Comment:
This must be related to my other [comment](https://github.com/WordPress/gutenberg/pull/72121/files#r2411200586), right?